File: jpa-example/src/main/java/example/jpa/model/Author.java
package example.jpa.model;

import example.jpa.model.listener.AuthorRatingCalculatingListener;
import jakarta.persistence.Entity;
import jakarta.persistence.EntityListeners;
import jakarta.persistence.OneToMany;
import jakarta.persistence.Transient;
import lombok.Getter;
import lombok.Setter;

import java.util.HashSet;
import java.util.Set;

@Getter
@Setter
@Entity
@EntityListeners(AuthorRatingCalculatingListener.class)
public class Author extends SerialId {
    private String name;

    @Transient
    private Double rating;

    @OneToMany(mappedBy = "author")
    private Set<Book> books = new HashSet<>();
}

File: jimmer-example/src/main/kotlin/example/jimmer/model/Author.kt
package example.jimmer.model

import example.jimmer.model.resolver.AuthorBestBookResolver
import example.jimmer.model.resolver.AuthorRatingResolver
import org.babyfish.jimmer.sql.Entity
import org.babyfish.jimmer.sql.OneToMany
import org.babyfish.jimmer.sql.Transient

@Entity
interface Author : SerialId {
    val name: String

    @Transient(AuthorRatingResolver::class)
    val rating: Double?

    @Transient(AuthorBestBookResolver::class)
    val bestBook: Book?

    @OneToMany(mappedBy = "author")
    val books: List<Book>
}